src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -566,86 +575,84 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
-                        response.ReasonPhrase = responseHeader.Substring(reasonPhraseIndex, reasonPhraseEnd - reasonPhraseIndex);
-                    }
-                    state._isRedirect = state._handler.AutomaticRedirection &&
-                         (response.StatusCode == HttpStatusCode.Redirect ||
-                         response.StatusCode == HttpStatusCode.RedirectKeepVerb ||
-                         response.StatusCode == HttpStatusCode.RedirectMethod) ;
-                }
+                statusCode = statusCode * 10 + (responseHeader[index] - '0');
+                index++;
             }
+            ResponseFormatAssert(statusCode >= 100 && statusCode < 600);
 
-            return true;
-        }
-
-        private static bool TryParseStatusCode(string responseHeader, int statusCodeStartIndex, out int statusCode)
-        {
-            if (statusCodeStartIndex + StatusCodeLength > responseHeader.Length)
+            //skip space
+            while(index < responseHeader.Length &&  responseHeader[index] == ' ')
             {
-                statusCode = 0;
-                return false;
+                index++;
             }
 
-            char c100 = responseHeader[statusCodeStartIndex];
-            char c10 = responseHeader[statusCodeStartIndex + 1];
-            char c1 = responseHeader[statusCodeStartIndex + 2];
+            // need reason phrase
+            ResponseFormatAssert( index < responseHeader.Length);
 
-            if (c100 < '0' || c100 > '9' ||
-                c10 < '0' || c10 > '9' ||
-                c1 < '0' || c1 > '9')
+            // Check if version is  HTTP/1.1 or HTTP/1.0
+            if (majorVersion == 1)
             {
-                statusCode = 0;
-                return false;
+                response.Version =
+                    minorVersion == 1 ? HttpVersion.Version11 :
+                    minorVersion == 0 ? HttpVersion.Version10 :
+                    new Version(0, 0);
+            }
+            else
+            {
+                response.Version = new Version(0, 0);
             }
 
-            statusCode = (c100 - '0') * 100 + (c10 - '0') * 10 + (c1 - '0');
+            response.StatusCode = (HttpStatusCode)statusCode;
+            response.ReasonPhrase = responseHeader.Substring(index);
+
+            state._isRedirect = state._handler.AutomaticRedirection &&
+                (response.StatusCode == HttpStatusCode.Redirect ||
+                 response.StatusCode == HttpStatusCode.RedirectKeepVerb ||
+                 response.StatusCode == HttpStatusCode.RedirectMethod) ;
 
             return true;
         }
 
+        private static bool ValidHeaderNameChar(char c)
+        {
+            // spec: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2
+            const string invalidChars = "()<>@,;:\\\"/[]?={}";
+            return c > ' ' && invalidChars.IndexOf(c) < 0 ;
```

Comment:
I see, ok, I misunderstood.

Why have the additional ```c > ' '``` rather than just adding ' ' and '\t' to the string being searched?  Seems to be optimizing for the failure case.